pkgs/development/beam-modules/elixir_ls.nix

```diff
@@ -0,0 +1,69 @@
+{ lib, elixir, fetchFromGitHub, fetchMixDeps, mixRelease }:
+let
+
+  pname = "elixir-ls";
+  version = "0.7.0";
+  src = fetchFromGitHub {
+    owner = "elixir-lsp";
+    repo = "elixir-ls";
+    rev = "f9ca65e05f13985c0a6270e6b9f4166f18a1357c";
+    sha256 = "0d0hqc35hfjkpm88vz21mnm2a9rxiqfrdi83whhhh6d2ba216b7s";
+    fetchSubmodules = true;
+  };
+
+  mixDeps = fetchMixDeps {
+    pname = "mix-deps-${pname}";
+    inherit src version;
+    sha256 = "0r9x223imq4j9pn9niskyaybvk7jmq8dxcyzk7kwfsi128qig1a1";
+  };
+
+in
+
+mixRelease {
+  inherit pname version src mixDeps;
+
+  # elixir_ls is an umbrella app
+  # override configurePhase to not skip umbrella children
+  configurePhase = ''
```

Comment:
configurePhase, buildPhase and installPhase are missing pre and post hooks.

Reply:
I'm not sure I understand this one.
Elixir Language Server is supposed to be a package and not a module, so I don't think we need to add pre and post hooks.
Let me know if I misunderstood something.

Reply:
It is recommended for derivations in general that override whole phases (instead of just using `<phase>Pre/Post`) to run these hooks even if those do not exist, it allows for using `<phase>Pre/Post` when you override a package. More info [here](https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md)

Reply:
Thanks!